src/Storage/Collection/LazyCollection.php

```diff
@@ -38,4 +26,23 @@ public function serialize()
 
         return $output;
     }
+
+    /**
+     * This overrides the base getIterator method and on access swaps out the EntityProxy objects
+     * for real entities.
+     *
+     * @return CallbackFilterIterator
+     */
+    public function getIterator()
+    {
+        $els = parent::getIterator();
+
+        return new CallbackFilterIterator($els, function (&$current) {
```

Comment:
ugh, filter is for filtering, not mapping... i'm sure there is a better way to do this

Reply:
like http://php.net/manual/en/function.iterator-apply.php

Reply:
@rossriley could you improve this?